### PR TITLE
Refactor to replace `style-search` for `value-no-vendor-prefix` rule

### PR DESCRIPTION
**Pull request by ybiquitous**
> Which issue, if any, is this issue related to?

- Part of #5499

> Is there anything in the PR that needs further explanation?

Replacing `style-search` with [`postcss-value-parser`](https://github.com/TrySound/postcss-value-parser).

### CODE REVIEW

File: lib/rules/value-no-vendor-prefix/index.js [hunk 1]
```diff
@@ -44,49 +56,47 @@ const rule = (primary, secondaryOptions, context) => {
 		}
 
 		root.walkDecls((decl) => {
+			const value = decl.value;
```
[the remark this answers is not in the document]
Comment (ybiquitous) [reply lib/rules/value-no-vendor-prefix/index.js:59]: Honestly, I have no strong opinion on restructuring.
But to keep consistency, ESLint's [prefer-destructuring](https://eslint.org/docs/rules/prefer-destructuring) rule will help it.

```diff
--- a/package.json
+++ b/package.json
@@ -64,6 +64,9 @@
       "require": true,
       "testRule": true
     },
+    "rules": {
+      "prefer-destructuring": "error"
+    },
     "root": true
   },
   "remarkConfig": {
```

```console
$ npm run lint:js
...

✖ 267 problems (267 errors, 0 warnings)
  208 errors and 0 warnings potentially fixable with the `--fix` option.
```

[the remark this answers is not in the document]
Comment (ybiquitous) [reply lib/rules/value-no-vendor-prefix/index.js]: When I try auto-fix of the `prefer-destructuring` rule, many cases are not fixed.
In my opinion, using ``prefer-destructuring`` may be hard. 🤔

[the remark this answers is not in the document]
Comment (jeddy3) [reply lib/rules/value-no-vendor-prefix/index.js]: Let's revisit enforcing it with ESLint.

**Review by jeddy3 — APPROVED** [review]
LGTM, thanks! 

Feel free to merge.

File: lib/rules/value-no-vendor-prefix/index.js [hunk 2]
```diff
@@ -44,49 +56,47 @@ const rule = (primary, secondaryOptions, context) => {
...
+					message: messages.rejected(node.value),
 					node: decl,
-					index: prop.length + (decl.raws.between || '').length + match.startIndex,
+					index: decl.prop.length + (decl.raws.between || '').length + node.sourceIndex,
 					result,
 					ruleName,
 				});
 			});
+
+			setDeclarationValue(decl, parsedValue.toString());
```
Comment (Mouvedia) [inline lib/rules/value-no-vendor-prefix/index.js:99]: @ybiquitous
Shouldn't this be surrounded by `if (needsFix) {}`?

**Pull request merged**